checker/src/abstract_value.rs

```diff
@@ -377,6 +377,40 @@ impl AbstractValueTrait for Rc<AbstractValue> {
         } else if self.is_top() {
             other
         } else {
+            if let Expression::Or { left: x, right: y } = &self.expression {
+                // (x || y) && x == x
+                // (x || y) && y == y
+                if *x == other || *y == other {
+                    return other;
+                }
+                if let Expression::LogicalNot { operand } = &other.expression {
+                    //((x || y) && (!x)) = y
+                    if *x == *operand {
+                        return y.clone();
+                    }
+                    //((x || y) && (!y)) = x
+                    if *y == *operand {
+                        return x.clone();
+                    }
+                }
+            }
+            match (&self.expression, &other.expression) {
```

Comment:
For my taste, beginning with this match this fn has become too large. Split?

Reply:
Not right now. It is nice to have everything in one place and it is too early conceptually differentiate things into sub categories that logically belong together and can be read in isolation.